Add service catalog tutorial (#3229)
wso2/docs-apim: en/docs/tutorials/integration-tutorials/service-catalog-tutorial.md

> # Publishing a sample API to Service Catalog

Reviewer: ```suggestion
# Exposing an Integration Service as a Managed API
```
This is the use case of the 'service catalog'. Therefore, shall we change the title to this
@samgnaniah

Reviewer 2: +1.

Author: Changed in a0c600c

## What you'll build

> In this tutorial we are re-creating the healthcare API we created in the [Sending a Simple Message to a Service](../sending-a-simple-message-to-a-service) tutorial, and publish that to the Service Catalog.

Reviewer 2: ```suggestion
In this tutorial we are re-creating the healthcare API we created in the [Sending a Simple Message to a Service]({{base_path}}/tutorials/integration-tutorials/sending-a-simple-message-to-a-service) tutorial, and publish that to the Service Catalog.
```

Please use base_path in the links. This is so that the links don't break if the structure changes and it's the strategy we use to consistently manage links in a sustainable manner.

Author: Fixed in a0c600c

#### Creating artifacts

Create **QueryDoctorEP** and **HealthcareAPI** by following the instructions given in [Create an Endpoint](../sending-a-simple-message-to-a-service/#create-an-endpoint) and [Create a REST API](../sending-a-simple-message-to-a-service/#create-a-rest-api) sections of **Sending a Simple Message to a Service** 
tutorial.

Once the **HealthcareAPI** is created, two new files with following names will be added to the metadata directory automatically.

1.  HealthcareAPI_metadata.yaml
2.  HealthcareAPI_swagger.yaml

<img src="{{base_path}}/assets/img/integrate/tutorials/service-catalog/metadata-folder-service-catalog.PNG" width="400">

!!! Note
    When registering a service in the Service Catalog, we need to provide the OpenAPI definition and metadata of the service we are registering. When creating an API, Integration Studio automatically creates these files for the user.

### Step 2: Configuring meta information

Open the HealthcareAPI_metadata.yaml and do the following changes.

> 1.  Change the **description** to explain what the API does. ( Ex API to fetch doctors of a given category )

Reviewer 2: ```suggestion
1.  Change the **description** to explain what the API does (e.g., API to fetch doctors of a given category).
```
The abbreviation "e.g." is actually the short form for a Latin term "exempli gratia". It means "for example". So when we use this abbreviation, we should always use the correct format or it would simply not mean the same thing. Please apply for other occurrences of this in this PR.

Author: ack

Author: Fixed in a0c600c

> 2.  Change serviceUrl from https to http. ( since HealthcareAPI is not secured )

Reviewer 2: ```suggestion
2.  Change `serviceUrl` from HTTPS to HTTP (since HealthcareAPI is not secured).
```

Author: Fixed in a0c600c

<img src="{{base_path}}/assets/img/integrate/tutorials/service-catalog/edit-metadata-service-catalog.PNG">

**serviceUrl** is the URL of the API when it gets deployed in the Micro Integrator. The API developer might not 
know this information in the development time. So, the URL can be parametrized to be resolved later using environment 
variables. 

We can configure the serviceUrl in following ways.

1.  Add the complete URL without parameters. Ex: http://localhost:8290/healthcare
2.  Parameterize using the host and port combination. Ex: http://{host}:{port}/healthcare
3.  Parameterize using a pre-configured URL. Ex: http://{url}/healthcare

We will configure the environment variables to resolve {host} and {port} configurations in an upcoming step. 

> Please [refer](link) to get more information on the other fields in the metadata YAML file.

Reviewer 2: The link is missing?

Reviewer 3: We should link this to the document where the format (fields) of the metadata file is described.

Author: Linked an empty page for this as a temporary fix. We will be adding the REST API docs later. Fixed in a0c600c.

You can subscribe and invoke the API by following the instructions given in the [Subscribe to an API]({{base_path}}/consume/manage-subscription/subscribe-to-an-api/) and [Invoke an API using the Integrated API Console]({{base_path}}/consume/invoke-apis/invoke-apis-using-tools/invoke-an-api-using-the-integrated-api-console/) documentation.

You will get the response message from the HealthcareService, if you send the category as `surgery`:

```json
[
    {
        "name":"thomas collins",
        "hospital":"grand oak community hospital",
        "category":"surgery",
        "availability":"9.00 a.m - 11.00 a.m",
        "fee":7000.0
    },
    {
        "name":"anne clement",
        "hospital":"clemency medical center",
        "category":"surgery",
        "availability":"8.00 a.m - 10.00 a.m",
        "fee":12000.0
    },
    {
        "name":"seth mears",
        "hospital":"pine valley community hospital",
        "category":"surgery",
        "availability":"3.00 p.m - 5.00 p.m",
        "fee":8000.0
    }
]
```

Now, check the **Console** tab of WSO2 Integration Studio and you will see the following message:
`INFO - LogMediator message = "Welcome to HealthcareService"`

> You have now created and deployed an API in API Manager which exposes the integration service created in the Micro Integrator, which receives requests, logs a message using the Log mediator, sends the request to a back-end service using the Send mediator, and returns a response to the requesting client.

Reviewer 2: ```suggestion
You have now created and deployed an API in API Manager that exposes the integration service created in the Micro Integrator. This receives requests, logs a message using the Log mediator, sends the request to a back-end service using the Send mediator, and returns a response to the requesting client.
```

Author: Fixed in a0c600c
